text/0054-kerberos-buildpack.md

```diff
@@ -0,0 +1,55 @@
+# Introduce Kerberos Buildpack
+
+## Summary
+
+Introduce a buildpack that installs the [MIT
+Kerberos](https://kerberos.org/dist/) libraries and executables such that
+applications authenticating with Kerberos can do so on any of the supported
+stacks.
+
+## Motivation
+
+Kerberos is currently supported for applications built on top of the Full
+stack. This is enabled through the addition of the `krb5-user` package in our
+stack definition. Unfortunately, the Full stack also includes a number of other
+packages that many application developers would like to avoid installing.
+
+Creating a new Kerberos buildpack will allow these developers to make a more
+focused addition to their application image be ensuring that only those
+packages relating to Kerberos are installed. It will also allow the project to
+support Kerberos on top of stacks that are not the Full stack without needing
+to also include that package in these other stack definitions.
+
+## Detailed Explanation
+
+A new repository will be created at `paketo-buildpacks/kerberos`. This repo
+will house a buildpack that has the singular purpose of installing the Kerberos
+libraries and executables into a layer made available to other buildpacks or at
+runtime. The buildpack will include Kerberos as a precompiled package that can
+be dropped into a layer with support for Jammy only. The buildpack will follow
+the existing provide/requires conventions, exposing a dependency called
+`kerberos`.
+
+## Rationale and Alternatives
+
+An alternative would be to add the `krb5-user` package to the Base stack
+images. Avoiding this would be preferrable as we should try not to increase the
+scope or vectors of attack for our most commonly used stack image.
+
+## Implementation
+
+The implementation will use all of the new dependency management infrastructure
+and so be contained to this one repository. The dependency will need to be
+compiled against a Jammy stack and hosted by ourselves as no precompiled
+distribution has been found.
+
+## Prior Art
+
+The most immediately similar example of this kind of library-installing
+buildpack is the ICU buildpack that is used within the .NET Core language
+family.
+
+## Unresolved Questions and Bikeshedding
+
+* Do we need to backport support to Bionic?
+* Should we explore using stack extensions to implement this?
```

Comment:
I am very interested in this, do you happen to have any insights into specific stack extension availability timelines? It looks like in the upstream the [issue to implement support](https://github.com/buildpacks/rfcs/issues/224) extensions is in progress. I wouldn't want to block this work on waiting for upstream completion, but I feel like this is a great candidate for that feature. I guess we can weigh whether its worth it to re-work this buildpack later with stack extensions.

Reply:
Stack extensions was the first thing that came to mind when I read this RFC. I get that we may not want to wait for that, but I think we should consider it as we have to a.) make this buildpack and b.) continue to support it. 

If extensions are fully available in 3-6 months and can do all this, it puts us in a bad spot. We have a.) invested engineer time & resources into the buildpack and b.) have to either very quickly pivot and deprecate this buildpack or continue supporting something that has been superseded by upstream features.

Reply:
I also wonder if it is sufficient to recommend apt-buildpack in the short term? Has that approach been tried? Even if it's not perfect, if it provides a workable solution now, then we could focus future efforts on using the stack extensions, which seems like the long term solution.

Reply:
> If extensions are fully available in 3-6 months and can do all this

To Ryan's point, it's also worth considering if lifecycle consumers (i.e. platform providers) will be ready to support this in 3-6 months. Does anyone have a window into how this is looking for `pack` or `kpack`?

Reply:
After witnessing a demo of the current stack extension implementation during the CNB Working Group meeting on 11/17, I feel it is insufficient in its current state to fulfill the requirements of this RFC. Specifically, we cannot "extend" a run image by running extra commands against the base image. Instead, we must produce an entirely new run image that can be referenced with a `FROM` directive. This means we would need to have new run images for each stack that just include the `krb5-user` library. Extrapolate this out for any other stack we might need to support in the future and the added maintenance burden becomes clear.

I will update the RFC with this context.